### PR TITLE
Refactor `<MessageInput /> component, update styles (bottom gradient, scrollbar)

### PR DESCRIPTION
**Pull request by ratik21**
### What does this do?

- refactored `<MessageInput>` component, moved it from `<ChatView>` (lower level) to `<MessengerChat>` (higher level in DOM).
- moved `reply` local state from container to global state in redux.
- added bottom gradient. 
- added background blur on composer
- messages can now be scrolled to the bottom of the composer.

image

image

### CODE REVIEW

File: src/components/chat-view-container/chat-view.test.tsx [hunk 1]
```diff
@@ -138,6 +128,7 @@ describe('ChatView', () => {
     expect(wrapper.find(InvertedScroll).hasClass('channel-view__inverted-scroll')).toBe(true);
   });
 
+  /*
```
Comment (dalefukami) [inline src/components/chat-view-container/chat-view.test.tsx:131]: Do we need this test any more?

File: src/components/chat-view-container/chat-view.tsx [hunk 2]
```diff
@@ -301,21 +289,11 @@ export class ChatView extends React.Component<Properties, State> {
           </div>
         )}
 
+        {/* i think we can remove the entire code below */}
```
Comment (dalefukami) [inline src/components/chat-view-container/chat-view.tsx:292]: Follow up PR next week?

**Review by dalefukami — APPROVED** [review]
Suggestion for next time: this could have been done in multiple smaller PRs. Eg: move the reply to global state, move send message up one level, etc.

File: src/components/chat-view-container/chat-view.test.tsx (hunk 1 above)
Comment (ratik21) [reply src/components/chat-view-container/chat-view.test.tsx:131]: removing.

File: src/components/chat-view-container/chat-view.tsx (hunk 2 above)
Comment (ratik21) [reply src/components/chat-view-container/chat-view.tsx:292]: yesssssssssss

**Pull request merged**